druid/src/widget/z_stack.rs

```diff
@@ -94,17 +91,22 @@ impl<T: Data> ZStack<T> {
 
 impl<T: Data> Widget<T> for ZStack<T> {
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut T, env: &Env) {
-        let is_pointer_event = matches!(
-            event,
-            Event::MouseDown(_) | Event::MouseMove(_) | Event::MouseUp(_) | Event::Wheel(_)
-        );
-
+        let mut previous_hot = false;
         for layer in self.layers.iter_mut() {
-            layer.child.event(ctx, event, data, env);
-
-            if is_pointer_event && layer.child.is_hot() {
-                ctx.set_handled();
+            if event.is_pointer_event() && previous_hot {
+                if layer.child.is_active() {
+                    ctx.set_handled();
```

Comment:
Do we still need this `set_handled`? Given your point about `set_handled` affecting the rest of the widget tree, it seems maybe better to avoid it?

Reply:
Do mean replace it with the `obstructed` flag or just remove it?

Reply:
There's a good chance I'm missing something, but I was thinking just remove it. Does that work?

I'm expecting that this child is active but not hot, and it will know how to correctly handle mouse events based on that, even if they aren't marked as handled.

Reply:
> I'm expecting that this child is active but not hot, and it will know how to correctly handle mouse events based on that, even if they aren't marked as handled.

The child only knows that it is not hot because it received a follow up event (Mouse event while active is set) which was marked as ´handled´.

Reply:
Can't the child do `ctx.is_hot()`?

Reply:
It is about the `Pod` not the widget. The `Pod` sees a mouse event inside its bounds. Therefore it sets its `hot` state to `true`. The only exception is that `active` and `hot` are both set. In that case the `Pod` sets `hot` to `false` but still propagates the event.

Reply:
You mean `handled` and not `active`, right? So in that case I understand why you need `set_handled` here, thanks for the explanation! I still think it isn't ideal because (as you explained in some other thread) it affects the global widget tree and not just the subtree under the zstack. But I don't think figuring it out needs to block this pr...

Reply:
Ups my mistake. I meant `active` and `set_handled`. In this case `hot` is set to `false` even if the curser is inside the widgets bounds

Reply:
> I still think it isn't ideal because (as you explained in some other thread) it affects the global widget tree and not just the subtree under the zstack. But I don't think figuring it out needs to block this pr...

I aggree. The only thing which is still brocken is scrolling while pressing a mousebutton. That is not the most impprtant usecase. I think we can merge this PR as it is.